### PR TITLE
WT-11153 Fix lz4 compressor to return error when the decoded length does not match uncompressed length

### PR DESCRIPTION
**Pull request by raviprakashgiri29**

### CODE REVIEW

File: ext/compressors/lz4/lz4_compress.c [hunk 1]
```diff
@@ -214,6 +214,18 @@ lz4_decompress(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, siz
         decoded = LZ4_decompress_safe((const char *)src + sizeof(LZ4_PREFIX), (char *)dst,
           (int)prefix.compressed_len, (int)dst_len);
 
+    /*
+     * In the majority of cases, when the decompressor successfully decompresses the entire source
+     * data, the decoded bytes matches the uncompressed length. However, there are some situations
+     * where the data is corrupted and the decompressor is unable to decompress the entire source
+     * data, in which case it differs and returns an error.
+     */
+    if (decoded != (int)prefix.uncompressed_len) {
```
Comment (quchenhao) [inline ext/compressors/lz4/lz4_compress.c:223]: Should we check against useful_len or uncompressed_len?

I think we should put this check inside if (decoded >= 0) section and cast decoded to (uint32_t)

Comment (raviprakashgiri29) [reply ext/compressors/lz4/lz4_compress.c:223]: @quchenhao I verified with the test run that `useful_len` and `uncompressed_len` are always the same, but as per the comments [here](https://github.com/wiredtiger/wiredtiger/blob/develop/ext/compressors/lz4/lz4_compress.c#L74) I am more inclined towards comparing against `prefix.useful_len`.

File: ext/compressors/lz4/lz4_compress.c [hunk 2]
```diff
@@ -215,6 +215,17 @@ lz4_decompress(WT_COMPRESSOR *compressor, WT_SESSION *session, uint8_t *src, siz
           (int)prefix.compressed_len, (int)dst_len);
 
     if (decoded >= 0) {
+        /*
+         * In the majority of cases, when the compressor successfully decompresses the entire source
+         * data, the decoded bytes matches the useful length. However, there are some situations
+         * where the data is corrupted and the compressor is unable to decompress the entire source
+         * data, in which case it differs and returns an error.
```
Comment (quchenhao) [inline ext/compressors/lz4/lz4_compress.c:222]: and we should return an error?

**Review by quchenhao — APPROVED** [review]
lgtm

**Comment by bvpvamsikrishna** [comment]
@raviprakashgiri29 , can we add the reproducer to this PR?

**Comment by raviprakashgiri29** [comment]
> @raviprakashgiri29 , can we add the reproducer to this PR?

@bvpvamsikrishna Did you mean to add the failed test/format config to the `failure_configs` folder?

**Comment by raviprakashgiri29** [comment]
> > @raviprakashgiri29 , can we add the reproducer to this PR?
> 
> @bvpvamsikrishna Did you mean to add the failed test/format config to the `failure_configs` folder?

Fixed in 70e58aae73d843de835cd27611ec722bc931dfa7

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**